src/libraries/System.Private.CoreLib/src/System/Environment.Win32.cs

```diff
@@ -56,7 +56,9 @@ private static void SetEnvironmentVariableFromRegistry(string variable, string?
                 {
                     IntPtr unused;
                     IntPtr r = Interop.User32.SendMessageTimeout(new IntPtr(Interop.User32.HWND_BROADCAST), Interop.User32.WM_SETTINGCHANGE, IntPtr.Zero, (IntPtr)lParam, 0, 1000, &unused);
-                    Debug.Assert(r != IntPtr.Zero, $"SetEnvironmentVariable failed: {Marshal.GetLastPInvokeError()}");
+
+                    // SendMessageTimeout message is a empty stub on Windows Nano Server that fails with both result and last error 0.
+                    Debug.Assert(r != IntPtr.Zero || Marshal.GetLastPInvokeError() == 0, $"SetEnvironmentVariable failed: {Marshal.GetLastPInvokeError()}");
```

Comment:
the docs say "If target is [User](https://learn.microsoft.com/en-us/dotnet/api/system.environmentvariabletarget?view=net-7.0#system-environmentvariabletarget-user) or [Machine](https://learn.microsoft.com/en-us/dotnet/api/system.environmentvariabletarget?view=net-7.0#system-environmentvariabletarget-machine), other applications are notified of the set operation by a Windows WM_SETTINGCHANGE message."

I don't know whether it's worth saying "except for Nano"..

Reply:
Nothing that uses HWND messages works on Nano Server (including e.g. `ApartmentState.STA`). This is just a corner case.

There are so many things that do not work on Nano Server. Documenting them would take some effort. We should wait for somebody asking for it first.